internal/pubsub/drivertest/drivertest.go

```diff
@@ -44,42 +44,40 @@ type Harness interface {
 	Close()
 }
 
-// HarnessMaker describes functions that construct a harness for running tests.
-// It is called exactly once per test; Harness.Close() will be called when the test is complete.
-type HarnessMaker func(ctx context.Context, t *testing.T) (Harness, error)
-
 // RunConformanceTests runs conformance tests for provider implementations of pubsub.
-func RunConformanceTests(t *testing.T, newHarness HarnessMaker) {
-	t.Run("TestSendReceive", func(t *testing.T) {
-		testSendReceive(t, newHarness)
-	})
-	t.Run("TestErrorOnSendToClosedTopic", func(t *testing.T) {
-		testErrorOnSendToClosedTopic(t, newHarness)
-	})
-	t.Run("TestErrorOnReceiveFromClosedSubscription", func(t *testing.T) {
-		testErrorOnReceiveFromClosedSubscription(t, newHarness)
-	})
-	t.Run("TestCancelSendReceive", func(t *testing.T) {
-		testCancelSendReceive(t, newHarness)
-	})
+//
+// newHarness is called exactly once per test to construct a Harness; its Close method
+// will be called when the test is complete.
+func RunConformanceTests(t *testing.T, newHarness func(context.Context, *testing.T) (Harness, error)) {
+	ctx := context.Background()
+	for _, test := range []struct {
+		name string
+		f    func(context.Context, *testing.T, Harness)
+	}{
```

Comment:
So I tried to follow this pattern before, but ended up getting rid of it in favor of the individual calls to `t.Run` because some of the `testXXX` functions ended up wanting to do multiple tests in a TDD way. For example:
https://github.com/google/go-cloud/blob/master/blob/drivertest/drivertest.go#L218

Also, some tests ended up needing slightly different arguments, e.g.:
https://github.com/google/go-cloud/blob/master/blob/drivertest/drivertest.go#L189

I suspect you will end up wanting the same (or at least, I don't think the pubsub drivertest coverage is good enough yet to be sure that you won't), so I still recommend we leave it alone at least for now.

Reply:
OK, closing.